src/components/e-input.vue

```diff
@@ -194,8 +212,8 @@
 
         return {
           ...this.stateModifiers,
-          notification: notification && this.hasFocus,
-          type: this.$attrs.type || 'text',
+          notification: notification && this.focus,
+          type: this.$attrs.type !== null || 'text',
```

Comment:
This is a different condition than before. The first part would resolve in a `boolean` (why did TS not warn about this?) also, the previous implementation would only use `$attrs.type` if it was truthy (non-empty sting, not null, not false, ...) and fall back to `text` if it wasn't. Now `type` will ALWAYS be either `true` or `'text'`.